### PR TITLE
soul-admin websocket listener testcase

### PR DESCRIPTION
**Pull request by hyuk-sudo**
Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: soul-admin/pom.xml [hunk 1]
```diff
@@ -169,6 +169,17 @@
             <version>1.4.200</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.undertow</groupId>
+            <artifactId>undertow-core</artifactId>
+            <version>2.2.2.Final</version>
```
Comment (tydhot) [inline soul-admin/pom.xml:175]: why add version here?

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+/**
+ * Data Change WebSocketListener Test.
+ *
+ * @author : Hyuk
+ * @description : WebsocketDataChangedListenerTest
+ * @date : 2020/12/6 9:31 下午
+ */
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class WebsocketDataChangedListenerTest {
```
Comment (tydhot) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java:67]: add final

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+import static io.undertow.Handlers.path;
+import static io.undertow.Handlers.websocket;
+
+/**
+ * Data Change WebSocketListener Test.
+ *
+ * @author : Hyuk
+ * @description : WebsocketDataChangedListenerTest
+ * @date : 2020/12/6 9:31 下午
```
Comment (tydhot) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java:63]: remove description and date

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+
+    /**
+     * start websocket client.
+     *
+     * @throws URISyntaxException URISyntaxException
+     * @throws InterruptedException InterruptedException
+     */
+    public void startClient() throws URISyntaxException, InterruptedException {
+        client = new WebSocketClient(new URI("ws://localhost:8888/websocket")) {
+            @Override
```
Comment (tydhot) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java:115]: the port can get by server's api, don't write fixed port

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+        authPathData.setEnabled(true);
+        authPathData.setPath(path);
+
+        List<AuthPathData> authPathDataList = new ArrayList<>();
+        authPathDataList.add(authPathData);
+        return authPathDataList;
+    }
+
+    /**
+     * 发送数据变更消息.
```
Comment (tydhot) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java:259]: remove chinese in java doc

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+     * test MetaData.
+     */
+    public void testOnMetaDataChanged() {
+        WebsocketData<MetaData> configData =
+                new WebsocketData<>(ConfigGroupEnum.META_DATA.name(), DataEventTypeEnum.CREATE.name(), metaDataList);
+        send(GsonUtils.getInstance().toJson(configData));
+    }
+
+    /**
+     * 处理webcosket消息进行分发.
```
Comment (tydhot) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketDataChangedListenerTest.java:326]: Chinese

**Review by tydhot — CHANGES_REQUESTED** [review]

File: soul-admin/pom.xml [hunk 7]
```diff
@@ -169,6 +169,17 @@
...
+        <dependency>
+            <groupId>io.undertow</groupId>
+            <artifactId>undertow-core</artifactId>
+            <version>${undertow.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.java-websocket</groupId>
+            <artifactId>Java-WebSocket</artifactId>
+            <version>1.5.0</version>
```
Comment (tydhot) [inline soul-admin/pom.xml:181]: I have a question about this.I think java-WebSockert  is imported by test case here, why not add test scope .And I think name the version as a property like undertow is better.

Comment (hyuk-sudo) [reply soul-admin/pom.xml:181]: sry, i forgot to test scope. The problem of version i will fix it with other pom file.

File: pom.xml [hunk 8]
```diff
@@ -118,6 +118,7 @@
         <curator-test.version>3.3.0</curator-test.version>
         <wiremock.version>2.18.0</wiremock.version>
         <jacoco-maven-plugin.version>0.8.5</jacoco-maven-plugin.version>
+        <java-websocket.version>1.5.0</java-websocket.version>
```
Comment (tydhot) [inline pom.xml:121]: I found java-websocket.version is defined too in soul-sync-data-websocket's pom.xml.Why not delete it in that file?

**Comment by tydhot** [comment]
There is some conflict files.May be you need pull the latest code to merge and push again.

File: pom.xml (hunk 8 above)
Comment (hyuk-sudo) [reply pom.xml:135]: fixed

**Review by tydhot — APPROVED** [review]

**Pull request merged**